### PR TITLE
[FrameworkBundle] Refactored assets:install command and apply Symfony styles

### PR DESCRIPTION
**Pull request by 1ed**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Replaces #13057

screenshot from 2015-04-06 17 26 47
screenshot from 2015-04-06 17 26 16
screenshot from 2015-04-09 10 24 59
screenshot from 2015-04-06 17 33 46
screenshot from 2015-04-06 20 14 57

### CODE REVIEW

**Comment by ogizanagi** [comment]
:+1:  IMO this is way better than your previous proposal. But there is still an issue (raised by @stof : https://github.com/symfony/symfony/pull/13057#discussion_r22383039 ) using the table layout, which doesn't allow to stream the output.
Can't we just use a list ?

However, I don't think streaming the output for this command is an absolute requirement.

**Comment by 1ed** [comment]
I was thinking about using a list too, but streaming the output with list is not possible for now either (at least using `SymfonyStyle`). I tried to steam the output using just `->sucess(), ->error(), ...`, but that was really ugly and unclear. I think the indentation provided by the table makes the output much more readable.

> And if any exception happens, there won't be any report about the symlinks which have already been created

Exceptions are caught and printed to the user.

**Comment by ogizanagi** [comment]
As we're currently loosing the paths information, do you think the verbosity should allow to add it in the bundle column ?

```
php app/console assets:install --symlink -v

// Trying to install assets as absolute symbolic links

--- --------------------------------------- ----------------------
     Bundle                                  Method / Error
--- --------------------------------------- ----------------------
 ✔︎   FrameworkBundle                        absolute symlink
     -> web/bundles/framework
 ✔︎   SensioDistributionBundle               absolute symlink
     -> web/bundles/sensiodistributions
--- --------------------------------------- ----------------------
```

**Comment by 1ed** [comment]
I'm not sure of using UTF-8 characters as they appearance in the console depends on a lot of things.

File: src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php [hunk 1]
```diff
@@ -11,20 +11,35 @@
 
 namespace Symfony\Bundle\FrameworkBundle\Command;
 
+use Exception;
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php]: This one should be removed and a \ added when needed in the code.

**Comment by sstok** [comment]
How about a progress bar to indicate running? You can show the table after its done (as summary) but having no output may cause users to think its stuck or something.

**Comment by fabpot** [comment]
This command should be really fast to execute, so I don't think fixing the "not streaming" problem is worth it.

File: src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php (hunk 1 above)
Comment (1ed) [inline src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php]: @fabpot done

**Comment by 1ed** [comment]
ping @fabpot

**Comment by 1ed** [comment]
@fabpot are there anything else I should do with this?

**Comment by fabpot** [comment]
:+1: for merge into 2.8.

**Comment by 1ed** [comment]
@javiereguiluz what do you think about this?

File: src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php [hunk 2]
```diff
@@ -74,77 +86,164 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+     * @throws IOException If link can not be created.
+     */
+    private function symlink($originDir, $targetDir, $relative = false)
+    {
+        if ($relative) {
+            $originDir = $this->filesystem->makePathRelative($originDir, realpath(dirname($targetDir)));
+        }
+        $this->filesystem->symlink($originDir, $targetDir);
+        if (!file_exists($targetDir)) {
+            throw new IOException(sprintf('Symbolic link "%s" is created but appears to be broken.', $targetDir), 0, null, $targetDir);
```
Comment (javiereguiluz) [inline src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php]: `... is created but ...` -> `... was created but ...` ?

**Comment by javiereguiluz** [comment]
:+1: thanks @1ed. Very nice work! Just solve the fabbot issue and we're good to merge :)

**Comment by 1ed** [comment]
Ready. Thank you!

**Comment by fabpot** [comment]
Thank you @1ed.

**Pull request closed without merging**